pkg/datastore/collectionstore.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import "context"
+
+type collectionStore struct {
+	col Collection
+	ds  DataStore
+}
+
+func (c *collectionStore) Find(ctx context.Context, opts ListOptions) (Iterator, error) {
+	return c.ds.Find(ctx, c.col.Kind(), opts)
```

Comment:
In this way, we still have to update the `Datastore` interface right?
Since this old Find function can not be used for FileDB.

Reply:
If so, from my point of view these two PRs are essentially the same.
And #3111 is just an extension of #3113 where `CollectionStore` interface is being added to be used by the caller as a helper.

Reply:
Yes, you're right. I'm about to use #3113 over this PR (I updated the Datastore interface in that PR). Tbh, the thing I worried about for that implementation is: `c.ds.Find(ctx, c.col, opts)` is a bit odd to me. Passing object properties to other object properties' method makes me feel like I can make a middle wrapper for that, but in that case, I have to make separated implementations for each `XXXStore` and maybe it's a bit overthinking, wdyt about that idea?

Reply:
Yes, making `XXXStore` is a bit overthinking to me as well.
I agree about using #3113 over this. 👍